compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/optimizer/BIROptimizer.java

```diff
@@ -67,6 +68,7 @@ private BIROptimizer(CompilerContext context) {
         this.rhsTempVarOptimizer = new RHSTempVarOptimizer();
         this.lhsTempVarOptimizer = new LHSTempVarOptimizer();
         this.lockOptimizer = new BIRLockOptimizer();
+        variableOptimizer = new BirVariableOptimizer();
```

Comment:
isn't it better to use "this.variableOptimizer" here? I always find it to be readable within a constructor

Reply:
For me it does not add more clarity(only more code) because already the IDE colors the global parameters differently so it is not at all hard to differentiate. I use `this` sparingly and only when needed.
I think it is a personal preference if it is not enforced. (https://stackoverflow.com/questions/5944382/is-it-clearer-to-always-use-this-or-only-when-necessary)

Reply:
However, if you look at the code above, we have followed a practice. The code you have added follows another practice. Isn't better to follow one single practice throughout?

Reply:
ack

Reply:
I would prefer without this for all 🙂